_source/_posts/2022-06-14-simple-crud-react-and-spring-boot.md

```diff
@@ -0,0 +1,1348 @@
+
+{% img blog/spring-boot-react/add-group.png alt:"Add Group screen" width:"800" %}{: .center-image }
+
+{% img blog/spring-boot-react/edit-group.png alt:"Edit Group screen" width:"800" %}{: .center-image }
+
+## Add authentication with Okta
+
+It's pretty cool to build a CRUD app, but it's even cooler to build a _secure_ one. To achieve that, you'll want to add authentication so users have to log in before viewing/modifying groups. To make this simple, you can use Okta's API for OIDC. At Okta, our goal is to make [identity management](https://developer.okta.com/product/user-management/) a lot easier, more secure, and more scalable than what you're used to. Okta is a cloud service that allows developers to create, edit, and securely store user accounts and user account data, and connect them with one or multiple applications. Our API enables you to:
+
+* [Authenticate](https://developer.okta.com/product/authentication/) and [authorize](https://developer.okta.com/product/authorization/) your users
+* Store data about your users
+* Perform password-based and [social login](https://developer.okta.com/authentication-guide/social-login/)
+* Secure your application with [multi-factor authentication](https://developer.okta.com/use_cases/mfa/)
+* And much more! Check out our [product documentation](https://developer.okta.com/documentation/)
+
+Are you sold? [Register for a forever-free developer account](https://developer.okta.com/signup/), and when you're done, come on back so you can learn more about building secure apps with Spring Boot!
+
+### Spring Security + OIDC
+
+[Spring Security added OIDC support in its 5.0 release](/blog/2017/12/18/spring-security-5-oidc). Since then, they've made quite a few improvements and simplified its required configuration. Add the Okta Spring Boot starter to do OIDC authentication.
+
+```xml
+<dependency>
+    <groupId>com.okta.spring</groupId>
+    <artifactId>okta-spring-boot-starter</artifactId>
+    <version>2.1.5</version>
+</dependency>
+```
+
+This dependency is a thin wrapper around Spring Security's OAuth and encapsulates the following dependencies:
+
+```xml
+<dependency>
+    <groupId>org.springframework.boot</groupId>
+    <artifactId>spring-boot-starter-security</artifactId>
+</dependency>
+<dependency>
+    <groupId>org.springframework.security</groupId>
+    <artifactId>spring-security-config</artifactId>
+</dependency>
+<dependency>
+    <groupId>org.springframework.security</groupId>
+    <artifactId>spring-security-oauth2-client</artifactId>
+</dependency>
+<dependency>
+    <groupId>org.springframework.security</groupId>
+    <artifactId>spring-security-oauth2-jose</artifactId>
+</dependency>
+```
+
+### Create an OIDC app in Okta
+
+{% include setup/cli.md type="web" framework="Okta Spring Boot Starter" signup="false"
+   loginRedirectUri="http://localhost:8080/login/oauth2/code/okta"
+   logoutRedirectUri="[http://localhost:3000,http://localhost:8080]" %}
```

Comment:
This displays as follows in the Preview: 
"Select Okta Spring Boot Starter. Then, change the Redirect URI to http://localhost:8080/login/oauth2/code/okta and use [http://localhost:3000,http://localhost:8080] for the Logout Redirect URI." 
Are those brackets as intended? I'm a little confused on what I need to edit.

Reply:
Yes, but it works with or without them (if the user copies them or not, it doesn't matter). I'll change to remove the brackets.